src/mlpack/tests/edge_boxes_test.cpp

```diff
@@ -0,0 +1,206 @@
+/**
+ * @file edge_boxes_test.cpp
+ * @author Nilay Jain
+ *
+ * Tests for functions in edge_boxes algorithm.
+ */
+
+#include <mlpack/core.hpp>
+#include <mlpack/methods/edge_boxes/feature_extraction.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "old_boost_test_definitions.hpp"
+
+using namespace mlpack;
+using namespace mlpack::structured_tree;
+/*
+  
+ 
+  //void GetShrunkChannels(CubeType& InImage, CubeType& reg_ch, CubeType& ss_ch);
+  
+  CubeType RGB2LUV(CubeType& InImage);
+
+  void Gradient(CubeType& InImage, 
+                MatType& Magnitude,
+                MatType& Orientation);
+
+  
+  CubeType Histogram(MatType& Magnitude,
+                       MatType& Orientation, 
+                       int downscale, int interp);
+ 
+  /*
+  CubeType ViewAsWindows(CubeType& channels, arma::umat& loc);
+
+  CubeType GetRegFtr(CubeType& channels, arma::umat& loc);
+
+  CubeType GetSSFtr(CubeType& channels, arma::umat& loc);
+
+  CubeType Rearrange(CubeType& channels);
+
+  CubeType PDist(CubeType& features, arma::uvec& grid_pos);
+  ***-/
+*/
+
+BOOST_AUTO_TEST_SUITE(EdgeBoxesTest);
+
+/**
+ * This test checks the feature extraction functions 
+ * mentioned in feature_extraction.hpp
+ */
+
+void Test(arma::mat m1, arma::mat m2)
+{
+  for (size_t i = 0; i < m1.n_cols; ++i)
+    BOOST_REQUIRE_CLOSE(m1(i), m2(i), 1e-2);  
+}
+
+void Test(arma::cube m1, arma::cube m2)
+{
+  for (size_t i = 0; i < m1.n_cols; ++i)
+    BOOST_REQUIRE_CLOSE(m1(i), m2(i), 1e-2);  
+}
+
+void DistanceTransformTest(arma::mat& input,
+                          double on, arma::mat& output,
+                          StructuredForests<arma::mat, arma::cube>& SF)
+{
+  arma::mat dt_output;
+  SF.DistanceTransformImage(input, on, dt_output);
+  Test(dt_output, output);
+} 
+
+void CopyMakeBorderTest(arma::cube& input,
+                              arma::cube& output,
+                          StructuredForests<arma::mat, arma::cube>& SF)
+{
+  arma::cube border_output;
+  SF.CopyMakeBorder(input, 1, 1, 1, 1, border_output);
+  Test(border_output, output);
+}
+
+void RGB2LUVTest(arma::cube& input, arma::cube& output,
+                  StructuredForests<arma::mat, arma::cube>& SF)
+{
+  double a, y0, maxi; 
+  a = std::pow(29.0, 3) / 27.0;
+  y0 = 8.0 / a;
+  maxi = 1.0 / 270.0;
+  arma::vec table(1064);
+  
+  for (size_t i = 0; i <= 1024; ++i)
+  {
+    table(i) = i / 1024.0;
+    
+    if (table(i) > y0)
+      table(i) = 116 * pow(table(i), 1.0/3.0) - 16.0;
+    else
+      table(i) = table(i) * a;
+
+    table(i) = table(i) * maxi;
+  }
+
+  for(size_t i = 1025; i < table.n_elem; ++i)
+    table(i) = table(i - 1);
+
+  arma::cube luv;
+  SF.RGB2LUV(input, luv, table);
+  Test(luv, output);
+}
+
+void ConvTriangleTest(arma::cube& input, int radius,
+    arma::cube& output, StructuredForests<arma::mat, arma::cube>& SF)
+{
+  SF.ConvTriangle(input, radius);
+  Test(input, output);
+}
+
+BOOST_AUTO_TEST_CASE(FeatureExtractionTest)
```

Comment:
Can you split the FeatureExtractionTest test case into smaller test cases? Something like:

```
BOOST_AUTO_TEST_CASE(DistanceTransformTest)
BOOST_AUTO_TEST_CASE(CopyMakeBorderTest)
```